pkg/activator/net/lb_policy.go

```diff
@@ -36,6 +36,39 @@ func randomLBPolicy(_ context.Context, targets []*podTracker) (func(), *podTrack
 	return noop, targets[rand.Intn(len(targets))]
 }
 
+// randomChoice2 implements the Power of 2 choices LB algorithm
+func randomChoice2(_ context.Context, targets []*podTracker) (func(), *podTracker) {
+	// Avoid random if possible.
+	l := len(targets)
+	// One tracker = no choice.
+	if l == 1 {
+		targets[0].addWeight(1)
+		return func() {
+			targets[0].addWeight(-1)
+		}, targets[0]
+	}
+	r1, r2 := 0, 1
+	// Two trackers - we know the both contestants,
+	// otherwise pick 2 random unequal integers.
+	if l > 2 {
+		r1, r2 = rand.Intn(l), rand.Intn(l)
+		for r1 == r2 {
```

Comment:
could probably skip the loop here by not including r1 in the choices for the second pick (ie rand(l), rand(l-1)). Seems fine to do this as a follow-on optimisation tho.

Reply:
You're up late :-)
I want to run this on the official benchmarks to see how it goes. We can try RR next (some variation thereof — least loaded RR).
But yeah we can probably squeeze a bit more from this, but it's gonna be nanoseconds.

Reply:
yeps, sgtm

Reply:
(early not late! :D)